lib/state.js

```diff
@@ -220,6 +230,41 @@ class State {
   }
 
   /**
+   * Begin a transaction
+   */
+  async beginTransaction() {
+    let client = await this.getClient();
+    try {
+      await client.query("BEGIN;");
+      return client;
+    } catch (err) {
```

Comment:
I think if an error is raised here and no client is returned, then there could be some unintended side effects, or at least the caller thinks it' operation on a transaction when it's really not.  beginTransaction will return undefined to the caller, and then that is passed onto other methods thinking it has a transaction.  In some cases, if undefined is passed in for the client, such as with _listTable, then it will just do a query without a transaction I think.  If we are explicitly wanting a transaction for an operation, we probably should verify that a client was returned and there was not an error thrown.